zebra-network/src/config.rs

```diff
@@ -37,19 +32,55 @@ pub struct Config {
 }
 
 impl Config {
-    fn parse_peers<S: ToSocketAddrs>(peers: HashSet<S>) -> HashSet<SocketAddr> {
-        peers
+    /// Concurrently resolves `peers` into zero or more IP addresses, with a timeout
+    /// of a few seconds on each DNS request.
+    ///
+    /// If DNS resolution fails or times out for all peers, returns an empty list.
+    async fn parse_peers(peers: &HashSet<String>) -> HashSet<SocketAddr> {
+        use futures::stream::StreamExt;
+        let peer_addresses = peers
             .iter()
-            .flat_map(|s| s.to_socket_addrs())
-            .flatten()
-            .collect()
+            .map(|s| Config::resolve_host(s))
```

Comment:
This line uses deref-coercion to convert `&String` to `&str`:
https://doc.rust-lang.org/book/ch15-02-deref.html
https://doc.rust-lang.org/std/string/struct.String.html#deref